src/machines/projectsMachine.ts

```diff
@@ -30,16 +51,20 @@ export const homeMachine = setup({
   actions: {
     setProjects: assign({
       projects: ({ context, event }) =>
-        'output' in event ? event.output : context.projects,
+        'output' in event && Array.isArray(event.output)
```

Comment:
Use the type guard I made in `utils.ts`.

```suggestion
        'output' in event && isArray(event.output)
```